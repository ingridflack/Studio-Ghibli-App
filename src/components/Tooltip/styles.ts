import styled, { css } from "styled-components";

export const Tooltip = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.fontSize.small};
    background-color: rgba(178, 7, 16, 0.8);

    color: ${theme.colors.gray.white};
    padding: 0.5rem 0.7rem;
    font-weight: 400;
    line-height: 1;

    border-radius: 5px;
    font-size: ${theme.fontSize.tiny};
    pointer-events: none;

    opacity: 0;
    position: absolute;
    bottom: 100%;
    left: 50%;
    transform: translateX(-50%);
    white-space: nowrap;
    transition: opacity 300ms ease;

    &::before {
      content: "";
      position: absolute;
      top: 100%;
      left: 50%;
      transform: translateX(-50%);
      border: 5px solid transparent;
      border-top-color: rgba(178, 7, 16, 0.8);
    }
  `}
`;

export const Container = styled.div`
  position: relative;

  &:hover {
    ${Tooltip} {
      opacity: 1;
    }
  }
`;
