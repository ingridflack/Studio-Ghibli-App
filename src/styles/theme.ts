const theme = {
  colors: {
    background: "#000",

    gray: {
      white: "#f6f6f6",
      platinum: "#e1e1e6",
      silver: "#a8a8b3",
      lightGray: "#7b8c98",
      dark: "#141414",
    },

    primary: "#e50914",
    secondary: "#b20710",
  },
  fontSize: {
    tiny: "0.65rem",
    small: "0.75rem",
    medium: "0.875rem",
    normal: "1rem",
    large: "1.5rem",
  },
} as const;

export type Theme = typeof theme;

export default theme;
