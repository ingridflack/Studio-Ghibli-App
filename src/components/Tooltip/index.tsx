import * as S from "./styles";

const Tooltip = ({ text, children, ...props }: any) => {
  return (
    <S.Container {...props}>
      <S.Tooltip>{text}</S.Tooltip>
      {children}
    </S.Container>
  );
};

export default Tooltip;
