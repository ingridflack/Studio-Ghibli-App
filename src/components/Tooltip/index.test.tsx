import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import Tooltip from ".";
import theme from "../../styles/theme";

describe("Tooltip", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Tooltip text="Test" />
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
