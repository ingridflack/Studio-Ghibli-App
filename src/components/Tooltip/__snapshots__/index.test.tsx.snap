// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should match snapshot 1`] = `
<div>
  <div
    class="sc-ade4b8be-1 kbXxyf"
  >
    <div
      class="sc-ade4b8be-0 jUqyaW"
    >
      Test
    </div>
  </div>
</div>
`;
